### DISCUSSION TITLE
Why the publish panel only can connect with ROS, how can i change it to support foxglove websocket server?

### DISCUSSION THREAD

**Discussion by chaosLeee** [foxglove/community#805] [Q&A] [answered]
I want to use studio publish a json to the foxglove websocket server ,but this time the studio only support ROS, I think its not difficult to change the studio to support websocket. waiting for answer.

**Comment by achim-k** [comment, marked as answer]
This is possible with every websocket server that implements the [Foxglove Studio WebSocket protocol](https://github.com/foxglove/ws-protocol/blob/main/docs/spec.md). You will be able to use the Publish panel if the server advertises the [`clientPublish`](https://github.com/foxglove/ws-protocol/blob/main/docs/spec.md#fields) capability.

You can find an example server that accepts JSON encoding here: https://github.com/foxglove/ws-protocol/blob/main/python/README.md
